### ISSUE TITLE
use `ref` instead of `tag` for git commits

### ISSUE THREAD

**Issue by aleklisi** [esl/MongoosePush#99]
```bash
$ MIX_ENV=prod mix do deps.get
* Getting pigeon (https://github.com/rslota/pigeon.git)
remote: Enumerating objects: 839, done.
remote: Total 839 (delta 0), reused 0 (delta 0), pack-reused 839
Receiving objects: 100% (839/839), 201.79 KiB | 0 bytes/s, done.
Resolving deltas: 100% (472/472), done.
warning: found quoted keyword "coveralls" but the quotes are not required. Note that keywords are always atoms, even when quoted. Similar to atoms, keywords made exclusively of Unicode letters, numbers, underscore, and @ do not require quotes
  /opt/MongoosePush/deps/pigeon/mix.exs:15
* Getting chatterbox (https://github.com/joedevivo/chatterbox.git)
remote: Enumerating objects: 2, done.
remote: Counting objects: 100% (2/2), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 2113 (delta 0), reused 2 (delta 0), pack-reused 2111
Receiving objects: 100% (2113/2113), 2.30 MiB | 3.19 MiB/s, done.
Resolving deltas: 100% (1502/1502), done.
* Getting sparrow (https://github.com/esl/sparrow.git)
fatal: reference is not a tree: b41ea2cc9e6c97ff5db784cb49402ec87234295d
** (Mix) Command "git --git-dir=.git checkout --quiet b41ea2cc9e6c97ff5db784cb49402ec87234295d" failed
$
```

https://github.com/esl/MongoosePush/blob/5b33e2fe8c90a66dc302608774d91fba0fb6315c/mix.exs#L31

tag -> ref

**Comment by aleklisi** [comment]
https://hexdocs.pm/mix/Mix.Tasks.Deps.html#module-git-options-git

**Comment by aleklisi** [comment]
solved by #100
